static/js/components/SaveCandidateButton.jsx

```diff
@@ -0,0 +1,210 @@
+import React, { useState, useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+import { useDispatch } from "react-redux";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import Checkbox from "@material-ui/core/Checkbox";
+import Button from "@material-ui/core/Button";
+import ButtonGroup from '@material-ui/core/ButtonGroup';
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
+import ClickAwayListener from "@material-ui/core/ClickAwayListener";
+import Grow from "@material-ui/core/Grow";
+import Paper from "@material-ui/core/Paper";
+import Popper from "@material-ui/core/Popper";
+import MenuItem from "@material-ui/core/MenuItem";
+import MenuList from "@material-ui/core/MenuList";
+import { useForm, Controller } from "react-hook-form";
+
+import * as sourceActions from "../ducks/source";
+import FormValidationError from "./FormValidationError";
+
+
+const options = ["Save as Source", "Save (select groups)"];
```

Comment:
For the first option, should say which group it will save to

Reply:
For the second option, the modal dialog should come up as soon as someone selects save to groups from the dropdown, rather than having to click the button after selecting the dropdown

Reply:
That's not how this kind of button works - think the github merge button. That behavior would require a major refactor. This seems to be a common enough type of button group, right? Especially for folks familiar with github pull requests.

Reply:
What you're asking for would best be provided by two separate buttons next to each other, if you feel strongly enough about not wanting to click the button type from the drop down and then click the button

Reply:
The drop down is just to change the button's action, not for actually clicking the button

Reply:
@dannygoldstein regarding your first comment, the button text will wrap to two lines if we add the group name to the button text. How about just a tooltip describing the quick save action?

Reply:
I thought of a way to implement your suggested dropdown button behavior, but still don't think that's desirable behavior from a UX perspective. Let's let others weigh in, though.

Reply:
@dannygoldstein I've updated https://github.com/skyportal/skyportal/issues/466 and request that we continue the conversation there and change it later if deemed necessary, resolving this for now.